tinygrad/codegen/uops.py

```diff
@@ -270,29 +270,33 @@ def uop_alu_idx(a: UOp, b, op, dtype=dtypes.int32):
     for loop_op in reversed([op for op in self.uops if op.uop is UOps.LOOP]):
       phis = set([u for u in self.get_recursive_children(loop_op) if u.uop is UOps.PHI])
       wheres = set([u for phi in phis for u in get_recursive_parents(phi) if u.arg == TernaryOps.WHERE])
-      if (any([u.uop not in allowed_ops or (u.uop is UOps.ALU and u.arg not in allowed_alus) for phi in phis for u in get_recursive_parents(phi)])
+      if (any([u.uop is not UOps.CONST for u in loop_op.vin])
+        or any([u.uop not in allowed_ops or (u.uop is UOps.ALU and u.arg not in allowed_alus) for phi in phis for u in get_recursive_parents(phi)])
         or any([where.vin[2].arg != 0 or where.vin[0].vin[1].uop is not UOps.CONST for where in wheres])
         or any(len([op for op in get_recursive_parents(where) if op.uop is UOps.LOOP]) == 0 for where in wheres)): continue
       if DEBUG >= 4 and (len(phis) > 0 or len(wheres) > 0): print("simplified {} PHI and {} WHERE in loop".format(len(phis), len(wheres)))
+      loop_length = loop_op.vin[1].arg - loop_op.vin[0].arg
+      for op in [op for op in self.uops if op.arg is BinaryOps.ADD]:
```

Comment:
eh don't do this, it creates the list first thus loops twice. maybe this?

```
for op in self.uops:
  if op.arg is BinaryOps.ADD and wheres.intersection(get_recursive_parents(op)) and phis.intersection(self.get_recursive_children(op)):
    op.vin = ...
```

Reply:
ok